webrtc.html

```diff
@@ -1471,39 +1484,42 @@ <h4>Set the RTCSessionDescription</h4>
                       <!-- D) transition stable to haveRemoteOffer -->
                       <li data-tests="RTCPeerConnection-setRemoteDescription-offer.html">
                         <p>If <var>description</var> is of type <code>"offer"</code>, set
-                        <var>connection</var>.<a>[[\PendingRemoteDescription]]</a>
-                        attribute to a new <code><a>RTCSessionDescription</a></code>
+                        <var>connection</var>'s <a>[[\PendingRemoteDescription]]</a>
+                        internal slot to a new <code><a>RTCSessionDescription</a></code>
                         object constructed from <var>description</var>, and set
                         <a>signaling state</a> to <code>"have-remote-offer"</code>.</p>
                       </li>
                       <!-- F) transition haveRemoteOffer or haveLocalProvAnswer to
                       stable -->
                       <li data-tests="RTCPeerConnection-setRemoteDescription-answer.html">
-                        <p>If <var>description</var> is of type <code>"answer"</code>, then
-                        this completes an offer answer negotiation. Set
-                        <var>connection</var>.<a>[[\CurrentRemoteDescription]]</a>
-                        to a new <code><a>RTCSessionDescription</a></code> object
-                        constructed from <var>description</var>, and set
-                        <var>connection</var>.<a>[[\CurrentLocalDescription]]</a>
-                        to <var>connection</var>.<a>[[\PendingLocalDescription]]</a>.
-                        Set both <var>connection</var>.<a>[[\PendingRemoteDescription]]</a>
-                        and <var>connection</var>.<a>[[\PendingLocalDescription]]</a>
-                        to <code>null</code>. Finally set <var>connection</var>'s
+                        <p>If <var>description</var> is of type <code>"answer"</code>,
+                        then this completes an offer answer negotiation. Set
+                        <var>connection</var>'s <a>[[\CurrentRemoteDescription]]</a>
+                        internal slot to a new <code><a>RTCSessionDescription</a></code>
+                        object constructed from <var>description</var>, and set
+                        <var>connection</var>'s <a>[[\CurrentLocalDescription]]</a>
+                        internal slot to <var>connection</var>'s
+                        <a>[[\PendingLocalDescription]]</a> internal slot.
+                        Set <var>connection</var>'s <a>[[\PendingRemoteDescription]]</a>
+                        and <a>[[\PendingLocalDescription]]</a> internal slots
+                        to <code>null</code>. Set <var>connection</var>'s
+                        <a>[[\RestartIce]]</a> internal slot to <code>false</code>,
```

Comment:
Above, at SLD(answer) there is a condition on restart: "if none of the current ICE credentials are present in `description`, then set `connection`'s `[[RestartIce]]` internal slot to `false`".
But here, at SRD(answer), it unconditionally clears the [[RestartIce]] slot. Was that an optimization? Shouldn't we check if ICE was restarted here too?

Otherwise I think the following would not cause an ICE restart:
```
await pc.setLocalDescription(await pc.createOffer());  // old credentials
pc.restartIce();
await pc.setRemoteDescription(answer from the previous offer);
```
Whereas I would expect this to trigger a follow-up negotiation.

In both SRD(answer) and SLD(answer) I think we should be explicit, rather than saying "if the current ICE credentials are different" we should compare the [[Current...]] description slots with the description before we update current to match.

Reply:
I'm implementing this right now... Question: in order to check if "current ICE credentials" are present in the description, does that mean I have to check both [[CurrentLocalDescription]] and [[CurrentRemoteDescription]] against description to be sure I don't miss any ufrags/pws?

Reply:
If I understand things correctly, this is what should be done at both SLD(answer) and SRD(answer):

Prior to updating the [[CurrentLocalDescription]] slot: if [[CurrentLocalDescription]] is null or none of its ICE credentials are present in _description_, set [[RestartIce]] to false.

Reply:
And ignoring rejected _description_ content in this comparison

Reply:
Sorry for the late response!

> Otherwise I think the following would not cause an ICE restart:

If we look at both *currentLocalDescription* and *pendingLocalDescription*, then I think this should work. I'll update the PR. 

Additionally, I'll update the PR to instead cache the descriptions at time of restartIce, and compare against those, to match Firefox.

If a local answer is being set, then compare against [[CurrentLocalDescription]].

If a remote answer is being set, then in the PR I had us always clear the [[RestartIce]] slot, thinking that this was the culmination of a negotiation round originating from our end. But as you point out, this does the wrong thing if `restartIce()` is called outside of stable. I think this means we want to compare against *currentRemoteDescription* and *pendingRemoteDescription*.

> And ignoring rejected description content in this comparison

I don't know that we need to skip rejected ones. Isn't rejection orthogonal?

Reply:
Discussed with @docfaraday. We can ignore remote ufrags thanks to JSEP, focusing on local ones and comparing them against the newly set currentDescription in SRD(answer).